crates/sdk/src/masp.rs

```diff
@@ -548,10 +620,15 @@ impl IntoIterator for Unscanned {
     type Item = IndexedNoteEntry;
 
     fn into_iter(self) -> Self::IntoIter {
-        self.txs.into_iter()
+        let txs = {
+            let mut txs: IndexedNoteData = Default::default();
+            let mut locked = self.txs.lock().unwrap();
+            std::mem::swap(&mut txs, &mut locked);
```

Comment:
Can't we just use `std::mem::take` here?

Reply:
yep, can fix.